File: start_ngrok.py
import subprocess
import time
import os

# Mata processos ngrok existentes
os.system('pkill -f ngrok 2>/dev/null')
time.sleep(1)

# Inicia ngrok
subprocess.Popen(['ngrok', 'http', '80'])
time.sleep(3)

# Verifica URL
try:
    import requests
    response = requests.get('http://localhost:4040/api/tunnels')
    data = response.json()
    
    for tunnel in data['tunnels']:
        if tunnel['proto'] == 'https':
            print(f"✓ Ngrok rodando: {tunnel['public_url']}")
            break
except:
    print("✓ Ngrok iniciado - verifique em http://localhost:4040")

File: start_ngrok.php
<?php
header('Content-Type: application/json');

if ($_SERVER['REQUEST_METHOD'] === 'POST') {
    // Script para iniciar ngrok
    $script = '#!/bin/bash
pkill -f ngrok
sleep 1
ngrok http 80 &
';
    
    file_put_contents('/tmp/start_ngrok.sh', $script);
    chmod('/tmp/start_ngrok.sh', 0755);
    
    // Executa o script
    exec('/tmp/start_ngrok.sh > /dev/null 2>&1 &');
    
    sleep(2);
    
    // Verifica se ngrok está rodando
    $check = exec('pgrep -f ngrok');
    
    if ($check) {
        echo json_encode(['success' => true, 'message' => 'Ngrok iniciado! Acesse http://localhost:4040 para ver a URL']);
    } else {
        echo json_encode(['success' => false, 'message' => 'Ngrok não iniciou. Verifique se está instalado.']);
    }
} else {
    echo json_encode(['success' => false, 'message' => 'Método não permitido']);
}
?>